src/app/recover-funds/[address]/RecoverFunds.tsx

```diff
@@ -0,0 +1,144 @@
+'use client';
+import { Address } from '@arbitrum/sdk';
+import { BigNumber, utils, constants } from 'ethers';
+import { useEffect, useState } from 'react';
+import { useNetwork } from 'wagmi';
+import { getProviderFromChainId, getTargetChainId } from '@/utils';
+import { supportedL1Networks } from '@/constants';
+import { RecoverFundsButton } from './RecoverFundsButton';
+
+interface OperationInfo {
+  balanceToRecover: BigNumber;
+  aliasedAddress: string;
+}
+
+const hasBalanceOverThreshold = (balanceToRecover: BigNumber) => {
+  // Aliased account will always have some leftover, we can't check for balance of 0, as it would always return 0
+  // We compare with 0.005 instead
+  return balanceToRecover.gte(BigNumber.from(5_000_000_000_000_000));
+};
+
+async function getData(
+  chainID: number,
+  address: string,
+): Promise<OperationInfo | null> {
+  // First, obtain the aliased address of the signer
+  const signerAddress = new Address(address);
+  const { value: aliasedAddress } = signerAddress.applyAlias();
+
+  // And get its balance to find out the amount we are transferring
+  try {
+    const l2Provider = getProviderFromChainId(chainID);
+    const aliasedSignerBalance = await l2Provider.getBalance(aliasedAddress);
+
+    return {
+      balanceToRecover: hasBalanceOverThreshold(aliasedSignerBalance)
+        ? aliasedSignerBalance
+        : constants.Zero,
+      aliasedAddress,
+    };
+  } catch (e) {
+    return {
+      balanceToRecover: BigNumber.from(100),
```

Comment:
Should this throw an error instead? What are the consequences of creating the OperationInfo object with this data?

Reply:
Good catch, changed to Zero